src/OneinchCallee.sol

```diff
@@ -0,0 +1,163 @@
+        require((z = x - y) <= x, 'ds-math-sub-underflow');
+    }
+
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _add(x, _sub(y, 1)) / y;
+    }
+
+    constructor(address daiJoinAddress) public {
+        daiJoin = DaiJoinLike(daiJoinAddress);
+        dai = daiJoin.dai();
+        dai.approve(daiJoinAddress, 2**256 - 1);
+    }
+
+    function _fromWad(address gemJoin, uint256 wad) internal view returns (uint256 amt) {
+        amt = wad / 10**(_sub(18, GemJoinLike(gemJoin).dec()));
+    }
+
+    function clipperCall(
+        address sender, // Clipper caller, pays back the loan
+        uint256 owe, // Dai amount to pay back          [rad]
+        uint256 slice, // Gem amount received           [wad]
+        bytes calldata data // Extra data, see below
+    ) external {
+        (
+            address to, // address to send remaining DAI to
+            address gemJoinAddress, // gemJoin adapter address
+            uint256 minProfit, // minimum profit in DAI to make [wad]
+            address router, // tx.to address received from the 1inch API
+            bytes memory oneinchData // reencoded 1inch parameters
+        ) = abi.decode(data, (address, address, uint256, address, bytes));
+
+        // Convert slice to token precision
+        slice = _fromWad(gemJoinAddress, slice);
+
+        // Exit gem to token
+        GemJoinLike(gemJoinAddress).exit(address(this), slice);
+
+        // Approve 1inch to take gem
+        TokenLike gem = GemJoinLike(gemJoinAddress).gem();
+        gem.approve(router, slice);
+        require(gem.balanceOf(address(this)) >= slice, 'gem-balance-too-low');
+
+        // Calculate amount of DAI to Join (as erc20 WAD value)
+        owe = _divup(owe, RAY);
+
+        uint256 oweWithProfit = _add(owe, minProfit);
+
+        OneinchRouter.SwapDescription memory swapDescription;
+        { // Build initial swapDescription
+            swapDescription.srcToken = gem;
+            swapDescription.dstToken = dai;
+            swapDescription.dstReceiver = address(this);
+            swapDescription.amount = slice;
+            swapDescription.minReturnAmount = oweWithProfit;
+        }
+
+        { // Execute 1inch swap
+            (
+                IAggregationExecutor executor,
+                address srcReceiver,
+                uint256 flags,
+                bytes memory permit,
+                bytes memory tradeData
+            ) = abi.decode(oneinchData, (IAggregationExecutor, address, uint256, bytes, bytes));
+
+            // Add missing swapDescription parameters
+            swapDescription.srcReceiver = srcReceiver;
+            swapDescription.flags = flags;
+
+            // Execute
+            OneinchRouter(router).swap(executor, swapDescription, permit, tradeData);
+            
+            // Check actual amount of dai returned
+            require(dai.balanceOf(address(this)) >= oweWithProfit, '1inch-returned-too-litle-dai');
```

Comment:
This can be outside the scope block if you want.  Not sure I have an opinion either way, other than you don't suffer the indentation and line-wrapping in an editooor if you move it outside the scope block.